File: API/Entities/PurchaseOrderLine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class PurchaseOrderLine
    {
        public int PurchaseOrderLineId { get; set; }
        public int PurchaseOrderId { get; set; }
        public int MasterPartId { get; set; }
        public string PartNumber { get; set; }
        public string PurchaseOrderLineNo { get; set; }
        public DateTime? RequestDate { get; set; }
        public DateTime? PromiseDate { get; set; }
        public DateTime? ReceivedDate { get; set; }
        public int OrderQty { get; set; }
        public string Notes { get; set; }

        public int ContainerId { get; set; }
        public Container Containers { get; set; }
        public PurchaseOrder PurchaseOrder { get; set; }
        public ICollection<ContainerDetail> ContainerDetails { get; set; }
    }
}

File: client/src/app/_models/purchaseOrderLine.ts
import { ContainerDetail } from "./containerDetail";

export interface PurchaseOrderLine {
    purchaseOrderLineId: number;
    purchaseOrderId: number;
    masterPartId: number,
    purchaseOrderLineNo: string;
    requestDate: Date;
    promiseDate: Date;
    receivedDate: Date;
    orderQty: number;
    notes: string;

    containerDetail: ContainerDetail[];
}